### PR TITLE
test/README: run go tests when updating images

### PR DESCRIPTION
**Pull request by achilleas-k**
Add a small paragraph with the instruction to run the go tests when pulling in a new version of images as a guide for making sure that osbuild-composer builds and runs successfully, before moving on to integration tests.

As a side note: This part of the document might be more appropriate in the HACKING document than the testing README but I'm not sure.  Or maybe somewhere else?  I'm not sure.

----

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Review by djach7 — APPROVED** [review]
Lgtm, very helpful info for contributors (like myself) who might not be super familiar in the repo.

File: test/README.md [hunk 1]
```diff
@@ -75,9 +75,16 @@ go mod edit -replace github.com/osbuild/images=github.com/<username>/images@<com
...
-tests against your updated code.
+Using the URL of the remote fork and branch will allow you to open a test PR
+and run the osbuild-composer integration tests against your updated code.
 
+Make sure you modify any code in osbuild-composer to adapt to the changes in
+the images repository. The images API is not considered stable and changes can
+occur frequently. A good quick check that everything compiles and _mostly_ runs
+correctly is by running the unit tests:
+```
+go test ./...
```
Comment (bcl) [inline test/README.md:86]: May as well also suggest `go vet ./...` which will catch basic things.

Comment (achilleas-k) [reply test/README.md:86]: Added!

**Review by thozza — APPROVED** [review]
Thanks!

**Pull request merged**